Classes/Units/Controllers/Inventory.js

```diff
@@ -54,7 +54,7 @@ class Inventory {
 		var Item = ItemList[itemId];
 		if(this.parent.stats.Gold < Item.GoldCost)
```

Comment:
Should be at line 79 I guess.
`if(this.parent.stats.Gold < effectiveGoldCost)`

Reply:
`function buildFromItems` need to be separated to estimate gold cost and remove items.
At the moment, we will need to have full GoldCost amount of item.
If this line will be moved, it will remove items but will not give us builded item if we won't have required gold.

Reply:
I need to boiler plate two times the `for loop`
I can rename function to something like `getEffectiveGoldCostAndRemoveBuildItems` to better reflect the function logic :D

In anycase, I can try to catch used slots in `buildFromItems` and execute them into the new function but I have to leave some _**dirty**_ instruction...

Reply:
function argument `remove` may be fine, it's not really a problem if it will be executed twice if it's necessary, buying item function is not executed simuntanously to care too much about performance